From 7439cc1da9a73844dfb47a39d42f89f918673f6f Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 8 Jun 2017 09:07:39 -0400
Subject: [PATCH] [1/2] CPU exhaustion in ReadDPXImage

Because dpx.file.image_offset is a unsigned int, it can be controlled
as large as 4294967295.
This will cause ImageMagick spend a lot of time to process a crafted
DPX imagefile, even if the imagefile is very small.

origin: https://github.com/ImageMagick/ImageMagick/commit/4faa884efefa85f4203291c8be2e6b0c2707554c
bug: https://github.com/ImageMagick/ImageMagick/issues/509
bug-debian: https://bugs.debian.org/867806

(cherry picked from commit 4faa884efefa85f4203291c8be2e6b0c2707554c)
---
 coders/dpx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/coders/dpx.c b/coders/dpx.c
index c1a4c7757..81a141370 100644
--- a/coders/dpx.c
+++ b/coders/dpx.c
@@ -1158,7 +1158,8 @@ static Image *ReadDPXImage(const ImageInfo *image_info,ExceptionInfo *exception)
            offset=SeekBlob(image,data_offset,SEEK_SET);
          else
            for ( ; offset < data_offset; offset++)
-             (void) ReadBlobByte(image);
+             if (ReadBlobByte(image) == EOF)
+               break;
           if (offset != data_offset)
             ThrowReaderException(CorruptImageError,"UnableToReadImageData");
        }
